src/sidebar/store/modules/annotations.js

```diff
@@ -388,6 +390,46 @@ function findAnnotationByID(state, id) {
   return findByID(state.annotations, id);
 }
 
+/**
+ * Return the number of page notes.
+ */
+const countNotes = createSelector(
+  state => state.annotations,
+  annotations => {
+    return arrayUtil.countIf(annotations, metadata.isPageNote);
+  }
+);
+
+/**
+ * Returns the number of annotations (as opposed to notes or orphans).
+ */
+const countAnnotations = createSelector(
+  state => state.annotations,
+  annotations => {
+    return arrayUtil.countIf(annotations, metadata.isAnnotation);
+  }
+);
+
+/**
+ * Returns the number of orphaned annotations.
+ */
+const countOrphans = createSelector(
+  state => state.annotations,
+  annotations => {
+    return arrayUtil.countIf(annotations, metadata.isOrphan);
+  }
+);
+
+/**
+ * Returns true if some annotations have not been anchored yet.
+ */
+const isWaitingToAnchorAnnotations = createSelector(
+  state => state.annotations,
+  annotations => {
+    return arrayUtil.countIf(annotations, metadata.isWaitingToAnchor) > 0;
```

Comment:
I'm guessing this was taken from existing code, but this could be written using just standard library functions as `annotations => annotations.some(metadata.isWaitingToAnchor)`